src/pd/pd.rs

```diff
@@ -486,20 +592,34 @@ impl<T: PdClient> Runner<T> {
         self.is_hb_receiver_scheduled = true;
     }
 
+    fn handle_reconnect(&mut self) {
+        if self.is_handle_reconnect_scheduled {
+            if self.invalidate_cache.swap(false, Ordering::Relaxed) {
+                info!("invalidate region heartbeat cache");
```

Comment:
what situation will the runner be re-run?

Reply:
And we need to reset `is_handle_reconnect_scheduled` here.

Reply:
When the client connects to a new PD leader.